redwood-layout-dom/src/commonMain/kotlin/app/cash/redwood/layout/dom/HTMLElementRedwoodLayoutWidgetFactory.kt

```diff
@@ -101,6 +105,27 @@ private class HTMLFlexContainer(
     value.overflowSetter(overflow.toCss())
   }
 
+  override fun onScroll(onScroll: ((Double) -> Unit)?) {
+    if (onScroll != null) {
+      val eventListener = scrollEventListener ?: object : EventListener {
+        override fun handleEvent(event: Event) {
+          val offset = when (value.style.flexDirection) {
+            "row" -> value.scrollTop
+            "column" -> value.scrollLeft
+            else -> throw AssertionError()
+          }
+          onScroll(offset)
+        }
+      }.also { scrollEventListener = it }
+      value.addEventListener("scroll", eventListener)
```

Comment:
I think this will add multiple listeners if you change the lambda. Probably need to move the remove code from the `else` branch to happen unconditionally, and then only if there's a non-null `onScroll` lambda do a set.

Reply:
I didn't check the view-based implementation, but if it's addXxx vs. setXxx it may also need the same treatment.

Reply:
I think we already handle this case? We keep a reference to the listener added in `addEventListener` and pass it back to `removeEventListener` if `onScroll` is back to null.

Reply:
The case I'm talking about is if the lambda changes its value.
```kotlin
var state by remember { mutableStateOf(false) }
val lambda = if (state) {
  { println("hey") }
} else {
  { println("hello") }
}
LaunchedEffect(Unit) {
  while (true) {
    delay(1.seconds)
    state = !state
  }
}

Column(onScroll = lambda)
```
This code will never send a null, but will send a lambda different from the existing one every second.

Reply:
Ah gotcha! Fixed